doc/source/groups_of_atoms.rst

```diff
@@ -0,0 +1,117 @@
+.. -*- coding: utf-8 -*-
+.. _groups-of-atoms:
+
+Groups of atoms
+===============
+
+MDAnalysis has a hierarchy of :class:`~MDAnalysis.core.groups.Atom` containers that are used throughout the code.
+
+.. image:: images/classes.png
+
+First and foremost is the :class:`~MDAnalysis.core.groups.AtomGroup`. An :class:`~MDAnalysis.core.groups.AtomGroup` is the primary :class:`~MDAnalysis.core.groups.Atom` container; virtually everything can be accessed through it, as detailed `above <https://www.mdanalysis.org/UserGuide/data_structures.html#atomgroup>`_. This includes chemically meaningful groups of :class:`~MDAnalysis.core.groups.Atom`\ s such as a :code:`Residue` or a :code:`Segment`. 
+
+---------------------
+Residues and Segments
+---------------------
+
+A :class:`~MDAnalysis.core.groups.Residue` is composed of :class:`~MDAnalysis.core.groups.Atom`\ s, and a :class:`~MDAnalysis.core.groups.Segment` is composed of :class:`~MDAnalysis.core.groups.Residue`\ s.
+
+These are considered atom :ref:`topology attributes <topology-attributes>`.
+
+The corresponding container groups are :class:`~MDAnalysis.core.groups.ResidueGroup` and :class:`~MDAnalysis.core.groups.SegmentGroup`. These have similar properties and available methods as :class:`~MDAnalysis.core.groups.AtomGroup`.
+
+Each of these container groups can be accessed through another. For example:
+
+.. code-block::
+
+    >>> ag = u.atoms.select_atoms('resname ARG and name CA')
+    >>> ag
+    <AtomGroup with 13 atoms>
+    >>> ag.residues
+    <ResidueGroup with 13 residues>
+    >>> ag.residues.atoms
+    <AtomGroup with 312 atoms>
+    >>> ag.segments
+    <SegmentGroup with 1 segment>
+    >>> ag.segments.atoms
+    <AtomGroup with 3341 atoms>
+
+Similarly, an :class:`~MDAnalysis.core.groups.Atom` has direct knowledge of the :class:`~MDAnalysis.core.groups.Residue` and :class:`~MDAnalysis.core.groups.Segment` it belongs to.
+
+.. code-block::
+
+    >>> a = u.atoms[0]
+    >>> a.residue
+    <Residue MET, 1>
+    >>> a.residue.segment
+    <Segment 4AKE>
+    >>> a.residue.segment.residues
+    <ResidueGroup with 214 residues>
+
+Adding to a topology
```

Comment:
Although this fits here, it's less important for someone starting out. I know it's hard to group things. Perhaps leverage the fact that you're writing hyperlinked docs and put it in a different section and link to it?

Reply:
The thrust of the previous section seemed to be on accessing and "reading" not on creating and "writing" topologies.

Reply:
Good idea, there is already a link to a relevant section in a notebook below as well.

Reply:
Moved this and Molecules to advanced_topology.rst for now